src/traces/box/plot.js

```diff
@@ -161,7 +161,8 @@ function plotPoints(sel, plotinfo, trace, t) {
     var bdPos = t.bdPos;
     var bPos = t.bPos;
 
-    var mode = trace.boxpoints;
+    // TODO ... unfortunately
+    var mode = trace.boxpoints || trace.points;
```

Comment:
We might want to rename `boxpoints` -> `points` as well as `boxmean` (maybe to  `showmean` + `showsd` ?)

Reply:
I thought about combining display attributes for box, mean line and standard dev line inside the violins in one flaglist attributes e.g. `stats: 'box+mean+sd'` or `innermode: /* */`, but when considering https://github.com/plotly/plotly.js/issues/1774 it might be best to have separate booleans e.g. `showbox`, `showmean`, `showsd` to keep things consisting with features like axis line and grids

So, unless someone opposes:

```js
showmeanline: true,
meanlinecolor: 'blue',
meanlinewidth: 1,
meanlinedash: 'dot'

showbox: true,
boxlinecolor: 'black'
boxlinewidth: 2,
boxfillcolor: 'red'
```

Reply:
Done in 6ffc379 with the help of `Lib.findPointOnPath` added in https://github.com/plotly/plotly.js/pull/2116/commits/4a40fc726cc42c161b2866a5f732b8c6b309c224